zipkin-common/src/test/scala/com/twitter/zipkin/storage/SpanStoreSpec.scala

```diff
@@ -258,4 +259,56 @@ abstract class SpanStoreSpec extends JUnitSuite with Matchers {
       Seq(Seq(span1))
     )
   }
+
+  /**
+   * Basic clock skew correction is something span stores should support, until
+   * the UI supports happens-before without using timestamps. The easiest clock
+   * skew to correct is where a child  appears to happen before the parent.
+   *
+   * It doesn't matter if clock-skew correction happens at storage or query
+   * time, as long as it occurs by the time results are returned.
+   *
+   * Span stores who don't support this can override and disable this test,
+   * noting in the README the limitation.
+   */
+  @Test def correctsClockSkew() {
```

Comment:
@dsyer @spencergibb fyi, this is good because it will break the zipkin-java project. I've a partial java impl of clock skew correction somewhere, and will finish that or ignore the test when bumping.